review-notifications/src/Github.js

```diff
@@ -131,52 +239,197 @@ class Github extends Component {
+                followedList={this.listOfFollowedPullRequests(2)}
+                user={this.state.user}
+                auth={this.state.auth}
+                token={this.state.token}
+                hasData={this.state.hasData}
+                classes={classes}
+              />
+            )}
+          />
+          <Route
+            path="/userRelated"
+            render={props => (
+              <UserRelatedPullRequests
+                {...props}
+                list={this.listOfPullRequests(userRelatedListSize)}
+              />
+            )}
+          />
+          <Route
+            path="/followed"
+            render={props => (
+              <FollowedPullRequests
+                {...props}
+                list={this.listOfFollowedPullRequests(followedListSize)}
+              />
+            )}
+          />
+        </Switch>
       </div>
     );
   }
 }
 
-export { Github };
+function MainSite(props) {
+  return (
+    <div>
+      {props.userRelatedList}
+      <Divider light />
+      <Grid item container>
+        <ListItem button component={RouterLink} to="/userRelated">
+          <ListItemText primary="Show more" />
+          <NavigateNextIcon />
+        </ListItem>
+      </Grid>
+      <Divider />
+      {props.followedList}
+      <Divider light />
+      <Grid item container>
+        <ListItem button component={RouterLink} to="/followed">
+          <ListItemText primary="Show more" />
+          <NavigateNextIcon />
+        </ListItem>
+      </Grid>
+      <Divider />
+      <Grid item container class={props.classes.bar}>
+        <ListItem
+          button
+          component="button"
+          onClick={() => chrome.runtime.openOptionsPage()}
+        >
+          <ListItemText primary="Options" />
+        </ListItem>
+      </Grid>
+    </div>
+  );
+}
+
+function UserRelatedPullRequests(props) {
+  return (
+    <div>
+      <Grid item container>
+        <ListItem button component={RouterLink} to="/">
+          <NavigateBeforeIcon />
+          <ListItemText primary="Back" />
+        </ListItem>
+      </Grid>
+      <Divider light />
+      {props.list}
+    </div>
+  );
+}
+
+function FollowedPullRequests(props) {
+  return (
+    <div>
+      <Grid item container>
+        <ListItem button component={RouterLink} to="/">
+          <NavigateBeforeIcon />
+          <ListItemText primary="Back" />
+        </ListItem>
+      </Grid>
+      <Divider light />
+      {props.list}
+    </div>
+  );
+}
+
+Github.propTypes = {
+  location: PropTypes.object.isRequired,
+  classes: PropTypes.object.isRequired,
+};
+
+MainSite.propTypes = {
+  classes: PropTypes.object.isRequired,
+  userRelatedList: PropTypes.object.isRequired,
+  followedList: PropTypes.object.isRequired,
+};
+
+UserRelatedPullRequests.propTypes = {
+  list: PropTypes.object.isRequired,
+};
+
+FollowedPullRequests.propTypes = {
+  list: PropTypes.object.isRequired,
+};
+
+export default withStyles(styles)(Github);
```

Comment:
The file is 435 lines long. Consider breaking it into a few smaller files/components. For example move the table rendering logic to a different component and use it here. Will make the code better organized and more readable.